sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobClientBase.java

```diff
@@ -261,23 +262,60 @@ public final BlobInputStream openInputStream(BlobRange range, BlobRequestConditi
+        ConcurrencyControl concurrencyControl = options.getConcurrencyControl() == null ? ConcurrencyControl.E_TAG
+            : options.getConcurrencyControl();
 
         BlobProperties properties = getProperties();
+        String eTag = properties.getETag();
+        String versionId = properties.getVersionId();
 
         BlobRange range = options.getRange() == null ? new BlobRange(0) : options.getRange();
         int chunkSize = options.getBlockSize() == null ? 4 * Constants.MB : options.getBlockSize();
 
         BlobRequestConditions requestConditions = options.getRequestConditions() == null
             ? new BlobRequestConditions() : options.getRequestConditions();
-        // Target the user specified version by default. If not provided, target the latest version.
-        if (requestConditions.getIfMatch() == null) {
-            requestConditions.setIfMatch(properties.getETag());
+        BlobAsyncClientBase client = this.client;
+
+        switch (concurrencyControl) {
+            case NONE:
+                if (requestConditions.getIfMatch() != null) {
+                    throw logger.logExceptionAsError(generateNoneException("requestConditions.ifMatch", "E_TAG"));
+                }
+                if (this.client.getVersionId() != null) {
+                    throw logger.logExceptionAsError(generateNoneException("client.versionId", "VERSION_ID"));
+                }
+                break;
+            case E_TAG:
+                // Target the user specified eTag by default. If not provided, target the latest eTag.
+                if (requestConditions.getIfMatch() == null) {
+                    requestConditions.setIfMatch(eTag);
+                }
+                break;
+            case VERSION_ID:
+                if (versionId == null) {
+                    throw logger.logExceptionAsError(
+                        new UnsupportedOperationException("Versioning is not supported on this account."));
+                } else {
+                    // Target the user specified version by default. If not provided, target the latest version.
+                    if (this.client.getVersionId() == null) {
+                        client = this.client.getVersionClient(versionId);
+                    }
+                }
+                break;
+            default:
+                throw logger.logExceptionAsError(
+                    new UnsupportedOperationException("Concurrency control type not supported."));
         }
 
         return new BlobInputStream(client, range.getOffset(), range.getCount(), chunkSize,
             requestConditions, properties);
     }
 
+    private UnsupportedOperationException generateNoneException(String wrongValue, String toSet) {
+        return new UnsupportedOperationException(String.format("'%s' can not be set when 'concurrencyControl'"
+            + " is set to NONE. Set 'concurrencyControl' to %s.", wrongValue, toSet));
```

Comment:
Should we make this into two constants as we only have two cases we need to cover and that will eliminate a string format.

Reply:
Yeah, this doesn't really need to be a method honestly

Reply:
With the addition of a few more error cases (if you take a second look), I think we should just leave this as a format.